java/src/main/java/science/icebreaker/network/keyword_merge/resolvers/MergeInDBResolver.java

```diff
@@ -0,0 +1,238 @@
+package science.icebreaker.network.keyword_merge.resolvers;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.neo4j.driver.Session;
+import org.neo4j.driver.Transaction;
+import org.neo4j.driver.TransactionWork;
+import org.neo4j.driver.internal.InternalNode;
+import org.neo4j.driver.Result;
+
+import science.icebreaker.network.Database;
+import science.icebreaker.network.Graph;
+import science.icebreaker.network.keyword_merge.comparators.SimilarityResult;
+import science.icebreaker.network.keyword_merge.entities.Keyword;
+import science.icebreaker.network.keyword_merge.repository.KeywordRepository;
+
+public class MergeInDBResolver implements SimilarityResultResolver {
+    /**
+     * Used for intermediate mapping. Ex: if keyword B is decided to be merged into
+     * keyword A and later on, Keyword C is decided to be merged into keyword B,
+     * this map can be used to map it to C instead
+     */
+    private Map<Keyword, Keyword> refersTo; // used for intermediate mapping
+    private KeywordRepository keywordRepo;
+
+    private PreparedStatement putKeywordRefStmt;
+    private PreparedStatement addNewKeywordStmt;
+    private PreparedStatement updateWeightStmt;
+
+    public MergeInDBResolver(KeywordRepository keywordRepo) {
+        this.keywordRepo = keywordRepo;
+        this.refersTo = new HashMap<Keyword, Keyword>();
+        try {
+            Connection conn = Database.getConnection();
+            putKeywordRefStmt = conn.prepareStatement(
+                "UPDATE keyword " 
+                + "SET refers_to=? "
+                + "WHERE name=?;"
+            );
+            addNewKeywordStmt = conn.prepareStatement(
+                "INSERT INTO keyword(name, weight) "
+                + "VALUES(?,0);"
+            );
+            updateWeightStmt = conn.prepareStatement(
+                "UPDATE keyword "
+                + "SET weight=? "
+                + "WHERE name=?;"
+            );
+        } catch (SQLException throwables) {
+            throw new RuntimeException(throwables);
+        }
+    }
+
+    public void merge(SimilarityResult result) {
+        if (!result.isSimilar)
+            return;
+
+        if (result.origin.equals(result.kw1)) {
```

Comment:
Some question: `Keyword.equals()` needs to be defined?

Reply:
Again, this was also working since similar strings keyword use the same ref.